components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/internal/utils/FlowUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.scim2.common.internal.utils;
+
+import org.wso2.carbon.identity.core.context.IdentityContext;
+import org.wso2.carbon.identity.core.context.model.Flow;
+
+/**
+ * Utility class for handling flow-related operations.
+ */
+public class FlowUtils {
+
+    /**
+     * Enters a flow with the specified flow name and initiating persona.
+     *
+     * @param flowName The name of the flow to enter.
+     */
+    public static void enterFlow(Flow.Name flowName) {
+
+        Flow.InitiatingPersona initiatingPersona = null;
+        if (IdentityContext.getThreadLocalIdentityContext().isApplicationActor()) {
+            initiatingPersona = Flow.InitiatingPersona.APPLICATION;
+        } else if (IdentityContext.getThreadLocalIdentityContext().isUserActor()) {
+            initiatingPersona = Flow.InitiatingPersona.ADMIN;
+        }
```

Comment:
Shall we have an else part and throw IllegalStateException() if actor is not set in the context for more safety

Reply:
That throw is there at the builder.